pkg/band/ism_2400.go

```diff
@@ -0,0 +1,147 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package band
+
+import (
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+)
+
+//revive:disable:var-naming
+
+var ism_2400 Band
+
+// ISM_2400 is the ID of the LoRa 2.4 GHz band.
+const ISM_2400 = "ISM_2400"
+
+//revive:enable:var-naming
+
+func init() {
+	defaultChannels := []Channel{
+		{
+			Frequency:   2403000000,
+			MaxDataRate: ttnpb.DATA_RATE_7,
+		},
+		{
+			Frequency:   2425000000,
+			MaxDataRate: ttnpb.DATA_RATE_7,
+		},
+		{
+			Frequency:   2479000000,
+			MaxDataRate: ttnpb.DATA_RATE_7,
+		},
+	}
+	const beaconFrequency = 2422000000
+
+	downlinkDRTable := [8][6]ttnpb.DataRateIndex{
+		{0, 0, 0, 0, 0, 0},
+		{1, 0, 0, 0, 0, 0},
+		{2, 1, 0, 0, 0, 0},
+		{3, 2, 1, 0, 0, 0},
+		{4, 3, 2, 1, 0, 0},
+		{5, 4, 3, 2, 1, 0},
+		{6, 5, 4, 3, 2, 1},
+		{7, 6, 5, 4, 3, 2},
+	}
+
+	ism_2400 = Band{
+		ID: ISM_2400,
+
+		MaxUplinkChannels: 16,
+		UplinkChannels:    defaultChannels,
+
+		MaxDownlinkChannels: 16,
+		DownlinkChannels:    defaultChannels,
+
+		SubBands: []SubBandParameters{
+			{
+				MinFrequency: 2400000000,
+				MaxFrequency: 2500000000,
+				DutyCycle:    1,
+				MaxEIRP:      8.0 + eirpDelta,
+			},
+		},
+
+		DataRates: map[ttnpb.DataRateIndex]DataRate{
+			ttnpb.DATA_RATE_0: makeLoRaDataRate(12, 812000, makeConstMaxMACPayloadSizeFunc(59)),
+			ttnpb.DATA_RATE_1: makeLoRaDataRate(11, 812000, makeConstMaxMACPayloadSizeFunc(123)),
+			ttnpb.DATA_RATE_2: makeLoRaDataRate(10, 812000, makeConstMaxMACPayloadSizeFunc(230)),
+			ttnpb.DATA_RATE_3: makeLoRaDataRate(9, 812000, makeConstMaxMACPayloadSizeFunc(230)),
+			ttnpb.DATA_RATE_4: makeLoRaDataRate(8, 812000, makeConstMaxMACPayloadSizeFunc(230)),
+			ttnpb.DATA_RATE_5: makeLoRaDataRate(7, 812000, makeConstMaxMACPayloadSizeFunc(230)),
+			ttnpb.DATA_RATE_6: makeLoRaDataRate(6, 812000, makeConstMaxMACPayloadSizeFunc(230)),
+			ttnpb.DATA_RATE_7: makeLoRaDataRate(5, 812000, makeConstMaxMACPayloadSizeFunc(230)),
+		},
+		MaxADRDataRateIndex: ttnpb.DATA_RATE_5,
```

Comment:
Why not 7?

Reply:
I don't think we should bother about this now, ADR will be disabled in ISM2400 since https://github.com/TheThingsIndustries/lorawan-stack/issues/2043

Reply:
Let's not set this at all then?